superduperdb/base/datalayer.py

```diff
@@ -181,9 +179,24 @@ def backfill_vector_search(self, vi, searcher):
             searcher.add(items)
             progress.update(len(items))
 
-    @property
-    def distributed_client(self):
-        return self._distributed_client
+    def set_compute(self, new: ComputeBackend):
+        """
+        Set a new compute engine
+        """
+        logging.warn(
+            f"Change compute engine from '{self.compute.name()}' to '{new.name()}'"
+        )
+
+        self.compute.disconnect()
+        logging.success(
+            f"Succesfully disconnected from compute engine: '{self.compute.name()}'"
+        )
+
+        logging.info(f"Connecting to compute engine: {new.name()}")
+        self.compute = new
```

Comment:
@blythed 
are we sure we want to change compute in an already running Datalayer instance

wouldn't this go against our discussion of NON Mutability of db instance

Reply:
There is no other way given the structure of the test-suite.

Reply:
if this is required in test-suite we can explicitly attach it in test-suite but I don't think we require it here

Reply:
In the test suite, don't we build another `db`?

Reply:
there is one `test_db` created at the top level, then is inherited by the `integration` tests, and it changes the `compute` layer.